Datascience_With_Python/Machine Learning/Videos/Elbow Method In KMeans Clustering/README.md

```diff
@@ -0,0 +1,94 @@
+K-means is a clustering algorithm which is centroid-based or we can say is based on the center point of each cluster, along with distance-based or we can say the distance between the cluster center explained above and points in a cluster , where we calculate the distances to assign a point to a cluster. In K-Means, each cluster is represented by a centroid or the cluster mean.  
+Now, to find number of clusters we will be using Elbow Method.
+
+![K means](./Images/K_Means.png)
+
+Example: Let us consider the example of a courier company which is currently having only one delivery center in a town/village. Now, the load on that delivery center is increasing as the parcels and letters are incoming and outgoing. The company wants to make few more centers so that they each center can target its own neighbourhood which is very near and doesn't need to interfere with parcels intended for other locality which is nearer to some other center. How can company do this? Here comes the k means clustering to rescue. The company will first see how many delivery center does it want in the town/village to form clusters that will be several areas and each cluster's centroid will be the delivery center. This finding of number of delivery centers is done with the help of Elbow method.
+
+For implementing KMeans, we use
+```
+from sklearn.clusters import KMeans
+``` 
+and to build model we use,
+
+```
+kmeans=KMeans(n_clusters=i,init='k-means++',max_iter=300,n_init=10)
+```
+
+Now to find this n_clusters, we use Elbow mwthod.
+
+## Elbow method
+The elbow method helps to choose the optimum value of ‘k’ (number of clusters) by fitting the model with a range of values of ‘k’. We calculate the sum of squared distance(SSD) of each object from its cluster center, and from that, we find the number until which sharp changes in SSD are occurring. Through this, are finding the number of clusters through which even after increasing the number of clusters, doesnt make a significant difference or in other words, we are finding clusters for which increasing cluster centers does not bring significant difference in SSD. This point seems like an elbow and is the required number of k we want, and thus the method is named as elbow method.
+#### Implementing Elbow Method
+
+Importing necessary libraries
+```
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+sns.set()
+import numpy as np
+from sklearn.cluster import KMeans
+```
+
+Reading csv file
+```
+data=pd.read_csv('california_cities.csv')
+data.head(10)
+```
+![Reading csv file](./Images/Read_Csv_Head.png)
+
+Plotting scatter plot for unclustered data
+
+```
+plt.scatter(data['longd'],data['latd'])
+plt.ylabel('Latitude')
+plt.xlabel('Longitude')
+plt.title('California cities latitude and longitude')
+plt.show()
+```
+![Scatter Plot 1](./Images/Scatter_1.png)
+
+Selecting latitude and longitude columns to apply k means 
+```
+x=data.iloc[:,1:3]
+x.head()
+```
+![Data 1](./Images/Data_Head_1.png)
```

Comment:
please don't add codes in your readme.

Reply:
@DhruvBajaj01 , I am just following contribution guidelines which are pinned on discord channel.
![Screenshot (1404)](https://user-images.githubusercontent.com/82665506/135767766-8c9a8c6f-63aa-4151-a99f-f66f73532674.png)
Plus, I do think that these code sections are better for understanding and will help someone if he/she wants to try to implement anything as from the video, it is not easy to copy and implement it.
Thankyou